massa-execution/src/vm.rs

```diff
@@ -267,8 +290,8 @@ impl VM {
                 |_| ReadOnlyResult::Error("Failed to run in read-only mode".to_string()),
                 |_| ReadOnlyResult::Ok,
             ),
-            // TODO: integrate with output events.
-            output_events: None,
+            // integrate with output events.
+            output_events: mem::take(&mut context.events).export(),
```

Comment:
why not have output_events be also an EventStore so that you can just take the whole event store ?

Reply:
and EventStore could have a method `EventStore::extend(&mut self, other: &EventStore)` to merge another event store into it while keeping a config-defined max number of elements.

This extend can then be used to extend the final EventStore with a history element that finalized

Reply:
I'm quite not sure about where to store final events: 
- the final ledger is in a ledger step in an execution context in the vm 
- for now I store the events in a step history item in the step history in the vm 
- when I query events, for now, I iter on step history items in vm methods
- from what I understood, I don't want/need to have access to past events from the execution context.

So I guess I can have a final EventStore in the VM, prune it every now and then, and updating it during run_final_step ? So upon query I'll have to iter on the step history items and on the final ones. (That means that I should return a Vec<(SCOutputEvent, bool)> instead of a Vec<SCOutputEvent> to indicate whether the event is final or not)

Reply:
> why not have output_events be also an EventStore so that you can just take the whole event store ?

That more a user interface issue, I don't know how people expect this to behave. I don't feel like returning a bunch of indexes on top of everything else through the api is really usefull, but I  may be wrong. @z80dev what do you expect from that ?